nautilus_core/model/src/data/bar.rs

```diff
@@ -163,10 +183,114 @@ impl BarType {
         spec: BarSpecification,
         aggregation_source: AggregationSource,
     ) -> Self {
-        Self {
+        Self::StandardBar {
+            instrument_id,
+            spec,
+            aggregation_source,
+        }
+    }
+
+    pub fn new_composite(
+        instrument_id: InstrumentId,
+        spec: BarSpecification,
+        aggregation_source: AggregationSource,
+
+        composite_instrument_id: InstrumentId,
+        composite_spec: BarSpecification,
+        composite_aggregation_source: AggregationSource,
+    ) -> Self {
+        Self::CompositeBar {
             instrument_id,
             spec,
             aggregation_source,
+
+            composite_instrument_id,
+            composite_spec,
+            composite_aggregation_source,
+        }
+    }
+
+    pub fn from_bar_types(standard_bar: &Self, composite_bar: &Self) -> Self {
```

Comment:
Following on from the above, `standard` and `composite` seems more aesthetic avoiding the underscore and is still clear enough?